search/binary.go

```diff
@@ -36,3 +36,45 @@ func BinaryIterative(array []int, target int, lowIndex int, highIndex int) (int,
 	}
 	return -1, ErrNotFound
 }
+
+//Returns index to the first element in the range [lowIndex, highIndex] that is not less than (i.e. greater or equal to) target, or return -1 and ErrNotFound if no such element is found.
+func LowerBound(array []int, target int, lowIndex int, highIndex int) (int, error) {
```

Comment:
In Go, we have access to the `len` function therefore it is unnecessary to have the low and high index in the function call. If a user wishes to do a subarray search, then they can pass the sub array like so `arr[:someIdx]` which changes the `len` I believe - have to check to make sure actually. But in any case, this could be simplified to this 😄 

 ```suggestion
func LowerBound(array []int, target int) (int, error) {
```

and use `highIndex := len(array) - 1` inside the function

Reply:
Actually, I agree with your suggestion. 
I write by this way because for following before code style as below.
`func Binary(array []int, target int, lowIndex int, highIndex int) (int, error)`
`func BinaryIterative(array []int, target int, lowIndex int, highIndex int) (int, error)`
Change all or keep this style. I'm a little confused, which dou you think is better.

Reply:
You can change the previous one as well. I think its best to be consistent.

Reply:
Except `func Binary`, all changed.  Pass the sub array like so `arr[:someIdx]`, that lowIndex and highIndex it's blong to `arr[:someIdx]` not `arr []int`. For example:
array := []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}
target := 10
the lowIndex is still `0` and highIndex is `len(arr[x:y])-1`, so the mid is not index in array []int.